core/src/main/java/org/elasticsearch/index/query/WildcardQueryBuilder.java

```diff
@@ -78,7 +106,60 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     }
 
     @Override
-    public String getName() {
-        return NAME;
+    protected Query doToQuery(QueryParseContext parseContext) throws IOException {
+        String indexFieldName;
+        BytesRef valueBytes;
+        
+        MappedFieldType fieldType = parseContext.fieldMapper(fieldName);
+        if (fieldType != null) {
+            indexFieldName = fieldType.names().indexName();
+            valueBytes = fieldType.indexedValueForSearch(value);
+        } else {
+            indexFieldName = fieldName;
+            valueBytes = new BytesRef(value);
+        }
+
+        WildcardQuery query = new WildcardQuery(new Term(indexFieldName, valueBytes));
+        MultiTermQuery.RewriteMethod rewriteMethod = QueryParsers.parseRewriteMethod(parseContext.parseFieldMatcher(), rewrite, null);
+        QueryParsers.setRewriteMethod(query, rewriteMethod);
+        return query;
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+        if (Strings.isEmpty(this.fieldName)) {
+            validationException = addValidationError("field name cannot be null or empty.", validationException);
+        }
+        if (this.value == null) {
```

Comment:
I think it would be interesting then to test what happens when you provide an empty string? is it something that we should accept or should we add validation for it?